RecoLocalTracker/SiStripClusterizer/plugins/alpaka/SiStripClusterizerConditionsESProducerAlpaka.cc

```diff
@@ -0,0 +1,278 @@
+      for (int j = 0; j < Data_apv_size; ++j) {
+        Data_apv_View.gain_(j) = gain[j];
+      }
+
+      return product;
+    }
+
+    // Auxiliary functions to translate indexes on the arrays
+    static constexpr uint16_t badBit = 1 << 15;
+    inline uint16_t fedIndex(uint16_t fed) { return (fed - FED_ID_MIN); };
+    inline uint32_t stripIndex(uint16_t fed, uint8_t channel, uint16_t strip) {
+      return (fedIndex(fed) * FEDCH_PER_FED * STRIPS_PER_FEDCH + channel * STRIPS_PER_FEDCH +
+              (strip % STRIPS_PER_FEDCH));
+    };
+    inline uint32_t apvIndex(uint16_t fed, uint8_t channel, uint16_t strip) {
+      return (fedIndex(fed) * APVS_PER_FEDCH * FEDCH_PER_FED + APVS_PER_CHAN * channel +
+              (strip % STRIPS_PER_FEDCH) / STRIPS_PER_APV);
+    };
+    inline uint32_t channelIndex(uint16_t fedId, uint8_t fedCh) { return (fedIndex(fedId) * FEDCH_PER_FED + fedCh); };
+
+  private:
+    edm::ESGetToken<SiStripGain, SiStripGainRcd> gainsToken_;
+    edm::ESGetToken<SiStripNoises, SiStripNoisesRcd> noisesToken_;
+    // Two tokens for quality because of the exc message:
+    // You may need multiple tokens if you want to get the same data in multiple transitions.
+    edm::ESGetToken<SiStripQuality, SiStripQualityRcd> qualityTokenA_;
+    edm::ESGetToken<SiStripQuality, SiStripQualityRcd> qualityTokenB_;
+
+    // Make conditions as in the RecoLocalTracker/SiStripClusterizer/plugins/ClustersFromRawProducer.cc module
+    void fillSiStripClusterizerConditions_(const SiStripQuality& quality,
+                                           const SiStripGain* gains,
+                                           const SiStripNoises& noises,
+                                           std::vector<float>& invthick,
+                                           std::vector<uint32_t>& detID,
+                                           std::vector<uint16_t>& iPair,
+                                           std::vector<uint16_t>& noise,
+                                           std::vector<float>& gain);
+  };
+
+  void SiStripClusterizerConditionsESProducerAlpaka::fillSiStripClusterizerConditions_(const SiStripQuality& quality,
+                                                                                       const SiStripGain* gains,
+                                                                                       const SiStripNoises& noises,
+                                                                                       std::vector<float>& invthick,
+                                                                                       std::vector<uint32_t>& detID,
+                                                                                       std::vector<uint16_t>& iPair,
+                                                                                       std::vector<uint16_t>& noise,
+                                                                                       std::vector<float>& gain) {
```

Comment:
here you could pass directly raw pointers (or event better `std::span`) to the SoA memory, and avoid the extra copy